src/textarea.rs

```diff
@@ -1879,10 +2036,163 @@ impl<'a> TextArea<'a> {
         scrolling.into().scroll(&mut self.viewport);
         self.move_cursor(CursorMove::InViewport);
     }
-}
 
+    // public functions for select
+
+    /// sets the style used for selected text
+    /// the default is LightBlue
+    ///
+    pub fn set_selection_style(&mut self, style: Style) {
+        self.select_style = style;
+    }
+
+    /// copies the current selection to the yank buffer
+    ///
+    pub fn copy(&mut self) {
+        if self.select_start.get().is_some() {
+            let (start, end, swapped) = self.normalize_selection();
+
+            self.cursor = start;
+            self.delete_range(start, end, true, false);
+            self.cursor = if swapped { start } else { end };
+        }
+    }
+
+    /// deletes the current selection
+    /// and places it in the yank buffer
```

Comment:
Can you add doc tests for each public APIs added in this PR so that users can understand the usage?